### PR TITLE
Fix the Gzip compression at the server side

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1899

The compression behaviour is as follows:
|Experiment|Client|Server|Request|Response|Status|
|---|---|---|---|---|---|
|1|GZIP disabled|GZIP disabled|Not compressed|Not compressed|Successful|
|2|GZIP enabled|GZIP disabled|Compressed|Not compressed|Successful|
|3|GZIP disabled|GZIP enabled|Not compressed|Compressed|Successful|
|4|GZIP enabled|GZIP enabled|Compressed|Compressed|Successful|

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/compression.bal [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+# Compression types that are supported.
+# GZIP - Gzip compression
+public enum CompressionType {
+   GZIP = "gzip"
+}
+
+# Enables the compression support by adding the `grpc-encoding` header to the given headers.
+# ```ballerina
+# map<string|string[]> headers = grpc:setCompression(GZIP);
```
Comment (daneshk) [inline ballerina/compression.bal:25]:
```suggestion
# map<string|string[]> headers = grpc:setCompression(grpc:GZIP);
```

File: native/src/main/java/io/ballerina/stdlib/grpc/ServerCall.java [hunk 2]
```diff
@@ -82,20 +79,16 @@ public void sendHeaders(HttpHeaders headers) {
...
-                        .map(mediaType -> mediaType.split("\\s*;\\s*")[0])
-                        .collect(Collectors.toList());
-                if (!acceptEncodings.contains(compressor.getMessageEncoding())) {
-                    // resort to using no compression.
-                    compressor = Codec.Identity.NONE;
-                }
+
+        if (headers != null && headers.contains(GrpcConstants.MESSAGE_ENCODING)) {
+            String messageEncodingType = headers.get(GrpcConstants.MESSAGE_ENCODING);
+            if (compressorRegistry.lookupCompressor(messageEncodingType) != null) {
```
Comment (dilanSachi) [inline native/src/main/java/io/ballerina/stdlib/grpc/ServerCall.java:85]:
```suggestion
            compressor = compressorRegistry.lookupCompressor(messageEncodingType);
            if (compressor == null) {
                compressor = Codec.Identity.NONE;
            }
```
Can we write it like this?

Comment (BuddhiWathsala) [reply native/src/main/java/io/ballerina/stdlib/grpc/ServerCall.java]: Added it in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/435/commits/3529e48c03f9df5acbf60c735abe9c2f2d5adfb3

**Review by daneshk — APPROVED** [review]
LGTM, does client handle the scenario where server compression doesn't support by the client.

**Comment by BuddhiWathsala** [comment]
Currently, it is somewhat difficult to test that since we only support Gzip. I'll check that as a separate task.

**Pull request merged**